ki.js

```diff
@@ -3,28 +3,21 @@
  * Copyright (c) 2015 Denis Ciccale (@tdecs)
  * Released under MIT license
  */
-!function (b, c, d, e) {
-
-  /*
-   * init function (internal use)
-   * a = selector, dom element or function
-   */
-  function i(a) {
-    c.push.apply(this, a && a.nodeType ? [a] : '' + a === a ? b.querySelectorAll(a) : e)
-  }
-
+!function (b, c, d) {
   /*
    * $ main function
    * a = css selector, dom object, or function
    * http://www.dustindiaz.com/smallest-domready-ever
    * returns instance or executes function on ready
    */
   $ = function (a) {
-    return /^f/.test(typeof a) ? /c/.test(b.readyState) ? a() : $(b).on('DOMContentLoaded', a) : new i(a)
-  }
+    return this instanceof $
+      ? c.push.apply(this, a && a.nodeType ? [a] : '' + a === a ? b.querySelectorAll(a) : d) // constructor
+      : /^f/.test(typeof a) ? /c/.test(b.readyState) ? a() : $(b).on('DOMContentLoaded', a) : new $(a) // function
```

Comment:
You can win 0 bytes minified / _1 byte gzipped_ with the following change, leveraging similar characters to optimize gzip algorithm. Not a big deal so I am not including it in the PR to not bring confusion, let me know if you want it :D

```suggestion
      : /nc/.test(typeof a) ? /c/.test(b.readyState) ? a() : $(b).on('DOMContentLoaded', a) : new $(a) // function
```

Reply:
interesting! I think is totally valid, in the end doesn't seem any less useful than checking for `^f`